markdown/m2h/handlers/dl.ts

```diff
@@ -37,19 +37,22 @@ export function asDefinitionList(h, node) {
     paragraph.children[0].value = paragraph.children[0].value.slice(
       DEFINITION_PREFIX.length
     );
+
+    const [firstDtChild, ...dtChildren] = all(h, {
+      ...node,
+      children:
+        terms.length == 1 && terms[0].type == "paragraph"
+          ? terms[0].children
+          : terms,
+    });
+    if (firstDtChild) {
+      dtChildren.unshift(
+        h(node, "a", { "data-link-to-id": "true" }, [firstDtChild])
+      );
```

Comment:
I don't think it is ideal that this modifies the Markdown engine. It would be better if the Markdown just produced a clean `<dl>`, and this extra feature were implemented on top of the HTML output.

Reply:
> I don't think it is ideal that this modifies the Markdown engine. It would be better if the Markdown just produced a clean `<dl>`, and this extra feature were implemented on top of the HTML output.

I don't disagree. @wbamberg Do you mind opening an issue for this?

Reply:
Fair point, I'll look into it.

Reply:
Thank you @KartikSoneji . I didn't say, but should have, that this is a really nice feature, and thank you for adding it :).

Reply:
Thanks!
Not sure if you remember but you reviewed it the first time around too: #3428